src/include/processor/operator/persistent/index_builder.h

```diff
@@ -20,7 +20,7 @@ using StringBuffer = common::StaticVector<std::pair<std::string, common::offset_
 
 class IndexBuilderGlobalQueues {
 public:
-    explicit IndexBuilderGlobalQueues(std::unique_ptr<storage::PrimaryKeyIndexBuilder> pkIndex);
+    explicit IndexBuilderGlobalQueues(storage::PrimaryKeyIndexBuilder* pkIndex);
```

Comment:
Curious, why can't the global queues have ownership of the index anymore? We have to be careful that accesses to the index in any other fashion are properly synchronized, right?

Reply:
Because RDF copy requires CopyNode and CopyRel in a single physical plan. And CopyRel needs to access `PrimaryKeyIndexBuilder` before it being flushed to disk. It's easier to access if the ownership is in CopyNode instead of GlobalQueue